MekHQ/src/mekhq/campaign/mission/Contract.java

```diff
@@ -391,88 +392,107 @@ public boolean test(int codePoint) {
 
-        // calculate overhead
-        switch (overheadComp) {
-            case OH_HALF:
-                overheadAmount = (long) (0.5 * c.getOverheadExpenses() * getLength());
-                break;
-            case OH_FULL:
-                overheadAmount = (long) (1 * c.getOverheadExpenses() * getLength());
-                break;
-            default:
-                overheadAmount = 0;
-        }
+    private int getLengthPlusTravel(Campaign c) {
+        int travelMonths = (int) Math.ceil(getTravelDays(c) / 30.0) * 2;
+        return getLength() + travelMonths;
+    }
 
-        // calculate support amount
-        if (c.getCampaignOptions().usePeacetimeCost() && c.getCampaignOptions().getUnitRatingMethod()
-                .equals(mekhq.campaign.rating.UnitRatingMethod.CAMPAIGN_OPS)) {
-            supportAmount = (long) ((straightSupport / 100.0) * c.getPeacetimeCost() * getLength());
+    public long getEstimatedTotalProfit(Campaign c) {
+        long profit = getTotalAmountPlusFeesAndBonuses();
+        profit -= c.getOverheadExpenses() * getLengthPlusTravel(c);
+        profit -= c.getMaintenanceCosts() * getLengthPlusTravel(c);
+        if (c.getCampaignOptions().usePeacetimeCost()) {
+            profit -= c.getPeacetimeCost() * getLengthPlusTravel(c);
         } else {
-            long maintCosts = 0;
-            for (Unit u : c.getUnits()) {
-                if (u.getEntity() instanceof Infantry && !(u.getEntity() instanceof BattleArmor)) {
-                    continue;
-                }
-                maintCosts += u.getWeeklyMaintenanceCost();
-            }
-            maintCosts *= 4;
-            supportAmount = (long) ((straightSupport / 100.0) * maintCosts * getLength());
+            profit -= c.getPayRoll() * getLengthPlusTravel(c);
         }
-
-        // calculate transportation costs
-        if (null != Planets.getInstance().getPlanetById(planetName)) {
-            transportAmount = (long) ((transportComp / 100.0) * 2 * c.calculateCostPerJump(false)
-                    * c.calculateJumpPath(c.getCurrentPlanet(), getPlanet()).getJumps());
-        }
-
-        // calculate transit amount for CO
-        if (c.getCampaignOptions().usePeacetimeCost() && c.getCampaignOptions().getUnitRatingMethod()
-                .equals(mekhq.campaign.rating.UnitRatingMethod.CAMPAIGN_OPS)) {
-            // contract base * transport period * reputation * employer modifier
-            transitAmount = (long) (c.getContractBase()
-                    * (((c.calculateJumpPath(c.getCurrentPlanet(), getPlanet()).getJumps()) * 2) / 4)
-                    * (c.getUnitRatingMod() * .2 + .5) * 1.2);
+		if(null != c.getPlanet(planetName) && c.getCampaignOptions().payForTransport()) {
+			JumpPath jumpPath = c.calculateJumpPath(c.getCurrentPlanet(), getPlanet());
+
+			boolean campaignOps = c.getCampaignOptions().useEquipmentContractBase();
+			profit -= 2 * c.calculateCostPerJump(campaignOps, campaignOps) * jumpPath.getJumps();
```

Comment:
Why multiply by 2?

Reply:
To follow the pattern elsewhere in Contract. I assume the reasoning there is that the transport payment pays for travel from a company's homeworld to the contract world, and from the contract world back to the homeworld.

That could well be wrong. If so, I can change that in this pull request, or it can be a follow-on pull request.